src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -115,6 +120,9 @@ def make_candidate_from_ican(
     def make_requirement_from_install_req(self, ireq):
         # type: (InstallRequirement) -> Requirement
         if ireq.link:
+            # TODO: Get name and version from ireq, if possible?
+            #       Specifically, this might be needed in "name @ URL"
+            #       syntax - need to check where that syntax is handled.
```

Comment:
Top-level requirements call this function directly. Does this comment also apply for them (even for the name+specifier form) as well?

Reply:
Yes, basically this comment applies as a general thing, I want to track down anywhere we can usefully work out a name/version and use them. But I'll look at that in a follow-up, the `InstallationCandidate` is the key one to get sorted for now.